platform/android/src/style/sources/custom_geometry_source.cpp

```diff
@@ -78,6 +80,17 @@ namespace android {
         peer.Call(*_env, cancelTile, (int)tileID.z, (int)tileID.x, (int)tileID.y);
     };
 
+    void CustomGeometrySource::releaseThreads() {
+        android::UniqueEnv _env = android::AttachEnv();
+
+        static auto releaseThreads = javaClass.GetMethod<void ()>(*_env, "releaseThreads");
+
+        assert(javaPeer);
```

Comment:
It is possible to remove a source from a map, which will drop the `javaPeer` to a weak reference, allowing that object to be destroyed. This should not be a hard assert, and this method should probably also be called when the source is removed from a map.

Reply:
Moved to #12551